electricpy/__init__.py

```diff
@@ -4823,148 +4989,184 @@ def ic_555_astable(R=None,C=None,freq=None,t_high=None,t_low=None):
-    The T attenuator consists of three resistors. Two of these are connected in 
-    series and the other one is connected from between the two other resistors to ground. 
+    The T attenuator is a type of attenuator that looks like the letter T.
+    The T attenuator consists of three resistors. Two of these are connected in
+    series and the other one is connected from between the two other resistors to ground.
     The resistors in series often have the same resistance.
 
     .. math:: R1 = Z0*(\frac{10^{\frac{A_{db}}{20}}-1}{10^{\frac{A_{db}}{20}}+1});
     .. math:: R2 = Z0*(\frac{10^{\frac{A_{db}}{20}}}{10^{\frac{A_{db}}{10}}-1})
     .. image:: /static/t-attenuator-circuit.png
 
     Parameters
-    ---------- 
+    ----------
     Adb: float Attenuation in db
     Z0: float Impedence
 
-    Returns 
-    ------- 
+    Returns
+    -------
     R1: float T attenuator R1
     R2: float T attenuator R2
     """
-    x = Adb/20
+    x = Adb / 20
+
+    R1 = Z0 * (_np.power(10, x) - 1) / (_np.power(10, x) + 1)
+    R2 = 2 * Z0 * _np.power(10, x) / (_np.power(10, 2 * x) - 1)
 
-    R1 = Z0*(_np.power(10, x)-1)/(_np.power(10, x)+1)
-    R2 = 2*Z0*_np.power(10, x)/(_np.power(10, 2*x)-1)
+    return R1, R2
 
-    return R1,R2
 
 def pi_attenuator(Adb, Z0):
     r"""
     Pi attenuator.
 
     The Pi attenuator is a type of attenuator that looks like the Greek letter π.
-    The Pi attenuator consists of three resistors. One of these is connected in series and 
+    The Pi attenuator consists of three resistors. One of these is connected in series and
     the other two are connected in parallel to ground. The parallel resistors often have the same resistance.
 
     .. math:: R1 = Z0*(\frac{10^{\frac{A_{db}}{20}}+1}{10^{\frac{A_{db}}{20}}-1})
     .. math:: R2 = \frac{Z0}{2}*(10^{\frac{A_{db}}{20}} - \frac{1}{10^{\frac{A_{db}}{20}}})
     .. image:: /static/pi-attenuator-circuit.png
 
     Parameters
-    ---------- 
+    ----------
     Adb: float Attenuation in db
     Z0: float Impedence
 
-    Returns 
-    ------- 
+    Returns
+    -------
     R1: float π attenuator R1
     R2: float π attenuator R2
     """
-    x = Adb/20
+    x = Adb / 20
+
+    R1 = Z0 * (_np.power(10, x) + 1) / (_np.power(10, x) - 1)
+    R2 = (Z0 / 2) * (_np.power(10, x) - (1 / (_np.power(10, x))))
+
+    return R1, R2
+
+
+def zener_diode(Vin_min: float, Vin_max: float, Vout: float, load_current: float):
+    r"""
+    Zener Dioded Resitance & Power Calculator.
+
+    A zener diode also allows the current to flow in the opposite direction
+    when the zener voltage is reached instead of exclusively letting flow from
+    the anode to the cathode. Zener diodes are used to supply reference voltages
+    and to prevent overvoltage.
+
+    The following formulas can be used to calculate the resistance,the power of
+    the resistor and the power of the zener diode.
+
+    Parameters
+    ----------
+    Vin_min: float Minimum input voltage in (volts)
```

Comment:
- [ ] visually align columns
- [ ] type should be on one line, description should be on following line

Reply:
# Alignment example:

```
Vin_min:      float
              Minimum input voltage in (volts)
Vin_max:      float
              Maximum input voltage in (volts)
Vout:         float
              Output voltage in (volts)
load_current: float 
              Load current in (amps)
```

and

```
R:          float
            Resitance of the zener diode
P_zener:    float
            Power of the zener diode
P_resistor: float
            Power of the resistor
```

This gives us visual appeal in the source code itself, and doesn't affect the Sphinx documentation negatively. It allows us to keep more consistent, appealing documentation directly in the code :)